src/controllers/IncapsulaController.ts

```diff
@@ -0,0 +1,183 @@
+import * as Boom from 'boom';
+import * as Hapi from 'hapi';
+import * as Joi from 'joi';
+import IMap from '../lib/IMap';
+import * as incapsula from '../lib/incapsula';
+import { buildErrorResponse, cleanupError } from '../lib/ResponseBuilder';
+import { ServiceResponse } from '../lib/service';
+import CacheController from './CacheController';
+
+export default class IncapsulaController implements CacheController {
+  public get flushZoneConfig(): Hapi.RouteOptions {
+    return {
+      handler: this.flushZone.bind(this),
+      tags: ['api', 'cache', 'incapsula'],
+      description: 'Flush Incapsula cache',
+      validate: {
+        params: this.paramsSchema,
+        payload: this.payloadSchema,
+        failAction: (_: Hapi.Request, __: Hapi.ResponseToolkit, error: Boom): Hapi.ResponseObject => {
+          throw cleanupError(error);
+        },
+      },
+      plugins: { 'hapi-swagger': { responses: this.responsesSchema } },
+    };
+  }
+
+  public async flushZone(request: Hapi.Request, _: Hapi.ResponseToolkit) {
+    try {
+      const response: ServiceResponse = await incapsula.flushSite(
+        request.params['zone_id'],
+        (request.payload as IMap<string>)['incapsulaApiID'],
+        (request.payload as IMap<string>)['incapsulaApiKey']
+      );
+      return { remoteStatusCode: 200, remoteResponse: response.body };
+    } catch (error) {
+      throw buildErrorResponse(error, 'incapsula');
+    }
+  }
+
+  public get flushURLsConfig(): Hapi.RouteOptions {
+    return {
+      handler: this.flushURLs.bind(this),
+      tags: ['api', 'cache', 'incapsula'],
+      description: 'Flush Incapsula cache URL',
+      validate: {
+        params: this.paramsSchema,
+        payload: this.payloadSchema.keys({
+          urls: Joi.array()
+            .items(
+              Joi.string().uri({
+                scheme: ['http', 'https'],
+              })
+            )
+            .min(1)
+            .required(),
+        }),
+        failAction: (_: Hapi.Request, __: Hapi.ResponseToolkit, error: Boom): Hapi.ResponseObject => {
+          throw cleanupError(error);
+        },
+      },
+      plugins: { 'hapi-swagger': { responses: this.responsesSchema } },
+    };
+  }
+
+  public async flushURLs(request: Hapi.Request, _: Hapi.ResponseToolkit) {
+    try {
+      await Promise.all(
```

Comment:
You should use a single API call to Incapsula as it support flushing multiple URL with a single call.